docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+```Patient``` class constructor to create a new ```Patient``` object based on the information stored back in step 1. 
+
+3. After which, it would then call the 
+```PatientList```’s ```getPatientList()``` command to get the ```List``` patient list object such that the ```Patient``` 
+object created beforehand can directly be inserted into the patient list. 
+
+4. After adding the patient into the patient list 
+object, the ```Storage```’s ```savePatientList()``` function will be called next so that the newly update list of 
+```patient``` is saved as offline data.  
+
+5. When the above operation is successful, it will call upon the ```Ui``` class’ 
+```showPatientAddSuccess()``` function to display the success of adding the ```Patient``` object into the patient list.
+
+If the supplied patient age is a word or is missing, the age will be set to ```-1```. This value is chosen to indicate 
+that there isn’t a valid age set. Thus, when displaying the age, if ```-1``` is encountered, show age as an empty string 
+instead.
+
+Below shows the sequence diagram for ```AddPatientCommand``` class
+
+![](images/AddPatientCommandSequence.png)
+ 
+
+##### 2.2.4.2 AddAppointmentClass
+
+To add an appointment, the ```AddAppointmentCommand``` class is used. For this ```AddAppointmentCommand``` class, it 
+serves as a façade class for the ```Main```, ```Appointment```, ```AppointmentList``` and the ```Storage``` class to 
+interact with one another. 
+
+![](images/AddAppointmentDiagram.png)
+
+1. Like the ```AddPatientCommand``` class, the ```AddAppointmentCommand``` object is first created by the ```Parser``` 
+object, where the information of the appointment is again stored in a Map that the ```AddAppoinmentCommand``` object would read from. 
+Said information will be stored in the ```AddAppoinmentCommand``` object
+
+2. When 
+the ```Main``` calls ```execute(Ui ui, Storage storage)```, the ```AddAppointmentCommand``` class would call upon the 
+```Appointment``` class to make an ```Appointment``` Object. 
+
+3. After which, the ```AddAppoinmentCommand``` object will 
+call upon the ```AppointmentList``` object to obtain the list of ```Appointments``` (get the ```List``` object that 
+represents the list of appointments by ```AppointmentList```’s ```getAppointmentList()``` command) so that it can 
+directly add the new ```Appointment``` object into the appointment list. 
+
+4. Finally, it will call upon the ```Storage``` 
+class’s ```saveAppoinmentList()``` function to save the updated appointment list. 
+
+5. Upon successfully adding the 
+```Appointment``` object into the appointment list, it will call upon the ```Ui``` class’ 
+```showAppointmentAddSuccess()``` function to display the success of adding the ```appointment``` into the appointment 
+list.
+
+Below shows the sequence diagram for ```AddAppointmentCommand``` class
+
+![](images/AddAppointCommandSequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/31772358/77993943-bee33180-735b-11ea-90d5-1fd6f561a897.png)

Same here: should there be a cut at the lower right corner of the loop rectangle?